nodebuilder/tests/sync_test.go

```diff
@@ -55,7 +63,22 @@ func TestSyncLightWithBridge(t *testing.T) {
 	h, err = light.HeaderServ.GetByHeight(ctx, 30)
 	require.NoError(t, err)
 
+	err = light.ShareServ.SharesAvailable(ctx, h.DAH)
+	assert.NoError(t, err)
+
+	for {
```

Comment:
As an alternative, this could use the new utils Retry method instead of a potentially infinite loop.

```go
err = test.Retry(100, 100*time.Millisecond, func() error {
   // Checking this first allows the duration between attempts to be the sleep
   if s.CatchUpDone {
      return nil
   }
   return fmt.Errorf("catch up not done; SharesAvailable error %v",light.ShareServ.SharesAvailable(ctx, h.DAH))
}
```

Reply:
Thanks. It would be nice, for sure. Although, we should go an even cleaner way - subscribing for the done event.

Reply:
As per [this](https://github.com/celestiaorg/celestia-node/pull/1341#discussion_r1019375142) tread